contrib/profiler/profiler.py

```diff
@@ -0,0 +1,366 @@
+
+
+# The analyze function
+def analyze_samples(sample_list):
+    count = len(sample_list)
+    min_cpu, min_cpu_idx = sample_list[0].get_cpu_usage(), 0
+    max_cpu, max_cpu_idx = sample_list[0].get_cpu_usage(), 0
+    min_gpu, min_gpu_idx = sample_list[0].get_gpu_usage(), 0
+    max_gpu, max_gpu_idx = sample_list[0].get_gpu_usage(), 0
+    min_mem, min_mem_idx = sample_list[0].get_mem_used(), 0
+    max_mem, max_mem_idx = sample_list[0].get_mem_used(), 0
+    max_read, max_read_idx = sample_list[0].get_read_bytes(), 0
+    sum_cpu, sum_gpu, sum_mem, sum_read = 0, 0, 0, 0
+    cpu_when_gpu_low = list()
+    mem_when_gpu_low = list()
+    disk_read_when_gpu_low = list()
+    for i in range(1, count):
+        if sample_list[i].get_cpu_usage() < min_cpu:
+            min_cpu = sample_list[i].get_cpu_usage()
+            min_cpu_idx = i
+        if sample_list[i].get_cpu_usage() > max_cpu:
+            max_cpu = sample_list[i].get_cpu_usage()
+            max_cpu_idx = i
+        if sample_list[i].get_gpu_usage() < min_gpu:
+            min_gpu = sample_list[i].get_gpu_usage()
+            min_gpu_idx = i
+        if sample_list[i].get_gpu_usage() > max_gpu:
+            max_gpu = sample_list[i].get_gpu_usage()
+            max_gpu_idx = i
+        if sample_list[i].get_mem_used() < min_mem:
+            min_mem = sample_list[i].get_mem_used()
+            min_mem_idx = i
+        if sample_list[i].get_mem_used() > max_mem:
+            max_mem = sample_list[i].get_mem_used()
+            max_mem_idx = i
+        if sample_list[i].get_read_bytes() > max_read:
+            max_read = sample_list[i].get_read_bytes()
+            max_read_idx = i
+
+        if sample_list[i].get_gpu_usage() <= 10:
+            cpu_when_gpu_low.append(sample_list[i].get_cpu_usage())
+            mem_when_gpu_low.append(sample_list[i].get_mem_used())
+            disk_read_when_gpu_low.append(sample_list[i].get_read_bytes())
+
+        sum_cpu += sample_list[i].get_cpu_usage()
+        sum_gpu += sample_list[i].get_gpu_usage()
+        sum_mem += sample_list[i].get_mem_used()
+        sum_read += sample_list[i].get_read_bytes()
+
+    length_gpu_low = len(cpu_when_gpu_low)
+    if length_gpu_low == 0:
+        length_gpu_low = 1
+
+    print('%f\t%f\t%f\t%f\t%f\t%f\t%f\t%f\t%f\t%f\t%f' %
+          (max_gpu, sum_gpu / count, max_cpu, sum_cpu / count, max_mem, sum_mem / count, max_read, sum_read,
+           sum(cpu_when_gpu_low) / length_gpu_low,
+           sum(mem_when_gpu_low) / length_gpu_low,
+           sum(disk_read_when_gpu_low) / length_gpu_low))
+    return [max_gpu, sum_gpu / count, max_cpu, sum_cpu / count, max_mem, sum_mem / count, max_read, sum_read,
+            sum(cpu_when_gpu_low) / length_gpu_low,
+            sum(mem_when_gpu_low) / length_gpu_low,
+            sum(disk_read_when_gpu_low) / length_gpu_low]
+
+
+def start_sample(container_id, period, analyze_period, one_duration, output_dir, gpu_id, *container_pid):
```

Comment:
Not quiet understand why use `*container_pid` here, I think use `container_pid` is enough . If you want to check if container_pid exist, you can set a default value such as -1 for container_pid, and check if the container_pid == -1 here